membership-attribute-service/app/services/IdentityAuthService.scala

```diff
@@ -1,41 +1,93 @@
+          identityId = defaultClaims.identityId,
+          username = defaultClaims.username,
+          emailValidated = unparsedClaims.getOptional[Boolean]("email_validated")
+        )
+    )
+
+  override def fromUser(user: User): MDAPIUserClaims = MDAPIUserClaims(
+    primaryEmailAddress = user.primaryEmailAddress,
+    identityId = user.id,
+    username = user.publicFields.username,
+    emailValidated = user.statusFields.userEmailValidated
+  )
+}
+class IdentityAuthService(apiConfig: IdapiConfig, oktaTokenVerifierConfig: OktaTokenVerifierConfig)(implicit ec: ExecutionContext) extends AuthenticationService {
 
   val idApiUrl = Uri.unsafeFromString(apiConfig.url)
 
-  val identityPlayAuthService = IdentityPlayAuthService.unsafeInit(idApiUrl, apiConfig.token, Some("membership"))
+  val identityPlayAuthService = {
+    val idapiConfig = IdapiAuthConfig(idApiUrl, apiConfig.token, Some("membership"))
+    IdentityPlayAuthService.unsafeInit(MDAPIUserClaimsParser,idapiConfig, oktaTokenVerifierConfig)
+  }
 
   def user(implicit requestHeader: RequestHeader): Future[Option[User]] = {
     getUser(requestHeader)
-      .map(user => Option(user))
       .handleError { err =>
-        if (err.isInstanceOf[UserCredentialsMissingError])
-          // IdentityPlayAuthService throws an error if there is no SC_GU_U cookie or crypto auth token
-          // frontend decides to make a request based on the existence of a GU_U cookie, so this case is expected.
-          SafeLogger.info(s"unable to authorize user - no token or cookie provided")
-        else
-          SafeLogger.warn(s"valid request but expired token or cookie so user must log in again - $err")
+        err match {
+          case UserCredentialsMissingError(_) =>
+            // IdentityPlayAuthService throws an error if there is no SC_GU_U cookie or crypto auth token
+            // frontend decides to make a request based on the existence of a GU_U cookie, so this case is expected.
+            SafeLogger.info(s"unable to authorize user - no token or cookie provided")
 
+           case OktaVerifierException(validationError: ValidationError) =>
+            validationError match {
+              case OktaValidationError(originalException) =>
+                SafeLogger.warn(s"could not validate okta token - $validationError", originalException)
+              case _ =>
+                SafeLogger.warn(s"could not validate okta token - $validationError")
+             }
+          case err =>
+            SafeLogger.warn(s"valid request but expired token or cookie so user must log in again - $err")
+        }
         None
       }
   }
 
-  private def getUser(requestHeader: RequestHeader): Future[User] =
-    identityPlayAuthService
-      .getUserFromRequest(requestHeader)
-      .map { case (_, user) => user }
-      .unsafeToFuture()
-
+  private def getUser(requestHeader: RequestHeader): Future[Option[User]] = {
+    val scopes = List("profile", "email")
```

Comment:
this is just an example in the real version we would have members-data-api specific scopes here, and we probably will have to make different calls with different scopes for different endpoints

Reply:
These will be passed through from client apps